### PR TITLE
Add docs pages to sitemap

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Speedup sitemap by not loading everything in memory
- Add non-track docs to general sitemap
- Add track docs to track sitemap

### CODE REVIEW

File: app/controllers/sitemaps_controller.rb [hunk 1]
```diff
@@ -71,17 +75,22 @@ def track
...
+      track.concepts.find_each do |concept|
         pages << [track_concept_url(track, concept), concept.updated_at, :monthly, 0.8]
       end
     end
 
-    track.exercises.active.each do |exercise|
+    pages << [track_docs_url(track.slug), track.updated_at, :monthly, 0.8]
+    track.documents.find_each do |doc|
+      pages << [track_doc_url(track.slug, doc.slug), doc.updated_at, :monthly, 0.8]
+    end
```
Comment (ErikSchierboom) [inline app/controllers/sitemaps_controller.rb]: Let me know what the important should be

File: app/controllers/sitemaps_controller.rb [hunk 2]
```diff
@@ -41,10 +41,14 @@ def general
       [blog_posts_url, BlogPost.published.pluck(:updated_at).last, :monthly, 0.95]
     ]
 
-    BlogPost.published.ordered_by_recency.each do |blog_post|
+    BlogPost.published.ordered_by_recency.find_each do |blog_post|
       pages << [blog_post_url(blog_post), blog_post.updated_at, :monthly, 0.75]
     end
 
+    Document.where(track: nil).find_each do |doc|
+      pages << [doc_url(doc.section, doc.slug), doc.updated_at, :monthly, 0.8]
```
Comment (ErikSchierboom) [inline app/controllers/sitemaps_controller.rb]: Let me know what the important should be

Comment (iHiD) [reply app/controllers/sitemaps_controller.rb]: `0.75`

File: app/controllers/sitemaps_controller.rb (hunk 1 above)
Comment (iHiD) [reply app/controllers/sitemaps_controller.rb]: `0.75`

File: app/controllers/sitemaps_controller.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ def index
       xml.sitemapindex(xmlns: "http://www.sitemaps.org/schemas/sitemap/0.9") do
         xml.sitemap { xml.loc sitemap_general_url(format: :xml) }
         xml.sitemap { xml.loc sitemap_profiles_url(format: :xml) }
-        Track.active.order('num_concepts DESC, num_exercises DESC').each do |track|
+        Track.active.order('num_concepts DESC, num_exercises DESC').find_each do |track|
```
Comment (iHiD) [inline app/controllers/sitemaps_controller.rb]: There's <1000 tracks

File: app/controllers/sitemaps_controller.rb [hunk 4]
```diff
@@ -41,10 +41,14 @@ def general
       [blog_posts_url, BlogPost.published.pluck(:updated_at).last, :monthly, 0.95]
     ]
 
-    BlogPost.published.ordered_by_recency.each do |blog_post|
+    BlogPost.published.ordered_by_recency.find_each do |blog_post|
```
Comment (iHiD) [inline app/controllers/sitemaps_controller.rb]: There's <1000 blog posts

**Pull request merged**